### PR TITLE
feat(ci): bootstrap initial workflows

### PR DESCRIPTION
**Pull request by jxs**
## Description

Sets up the initial configurations for the project:
- CI `rustfmt`, `clippy` testing and building 
- Issue and PR templates with respective labels to help triage 

## Notes

We should create an issue to add [`cargo-semver-checks`](https://github.com/obi1kenobi/cargo-semver-checks) when we start publishing, so CI catches possible breaking changes

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by tchataigner** [comment]
Thanks for the PR ! 

I've added you with _write_ persmissions on the [project management](https://github.com/orgs/polyphene/projects/1/views/1) that goes along this repo. feel free to create the new issue related to  [`cargo-semver-checks`](https://github.com/obi1kenobi/cargo-semver-checks) there :+1:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        with:
+          shared-key: msrv-cache
+          save-if: false
+
+      - uses: dtolnay/rust-toolchain@stable
+
+      - uses: Swatinem/rust-cache@359a70e43a0bb8a13953b04a90f76428b4959bb6 # v2.2.0
+        with:
+          shared-key: stable-cache
+          save-if: false
```
Comment (tchataigner) [inline .github/workflows/ci.yml:49]: From what I understand you're suggesting to cache the use of `rust-problem-matchers` and `rust-cache` ? Could we consider caching crates dependencies also ? Or maybe one of those already set does this for us ?

**Review by tchataigner — CHANGES_REQUESTED** [review]
Two quick questions before approving!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+## Description
+
+<!-- Please write a summary of your changes and why you made them.-->
+<!-- This section will appear as the commit message after merging. Please craft it accordingly. -->
+
+## Notes
+
+<!-- Any notes or remarks you'd like to make about the PR. -->
+
+## Links to any relevant issues
```
[the remark this answers is not in the document]
Comment (jxs) [reply .github/pull_request_template.md]: wdyt of combining them instead then? I have updated the PR with them combined, if you prefer I can then update again for separate sections.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jxs) [reply .github/workflows/ci.yml:49]: no no Thomas. So, `Swatinem/rust-cache` caches the cargo registry and the `target` dir see [here](https://github.com/Swatinem/rust-cache#example-usage) for more info, [`rust-problem-matchers`](https://github.com/r7kamura/rust-problem-matchers) helps showing to the user which line the problem occured:
image

Comment (tchataigner) [reply .github/workflows/ci.yml:49]: > Swatinem/rust-cache caches the cargo registry and the target dir

Ok then I'm not 100% sure to understand why we call it twice in the same job, once before installing rust stable and one after :thinking:

Comment (jxs) [reply .github/workflows/ci.yml]: ah thanks for noticing, it was a typo. We don't need the `mrsv` cache :grinning:

**Review by tchataigner — APPROVED** [review]
Thanks for that, looks awesome ! I like when the world is automated :robot:

**Pull request merged**